### ISSUE TITLE
Adding the authentication service's proto repository as a submodule.

### ISSUE THREAD

**Issue by mkaramuk** [rika-hq/rika-api-gateway#2]
[Auth service](https://github.com/rika-hq/rika-auth) has protobuf definitions at [here](https://github.com/rika-hq/rika-auth-proto).  This repository should be added here as a submodule so we can use the same proto files.

**Comment by mkaramuk** [comment]
Issue blocked by #1

**Comment by rasnesakam** [comment]
After done some research and lookup, I found gradle file that we want [here](https://github.com/grpc/grpc-java/blob/master/examples/build.gradle).  After applying this configuration, project defines proto file under main module.
<img width="236" alt="image" src="https://github.com/rika-hq/rika-api-gateway/assets/45129228/95939c91-2e30-4356-9d0b-dd0d03830ee8">
When I look at to [this video](https://www.youtube.com/watch?v=DU-q5kOf2Rc) Guy created proto folder inside main module, and defined it as a resource folder. Our is defined as source folder. This configuration also implements some tasks for proto generating like below, But I didn't tested it
<img width="271" alt="image" src="https://github.com/rika-hq/rika-api-gateway/assets/45129228/59a902f5-de7e-4eb7-95eb-65b53d75a330">
In my opinion, this case covers issue #3 .
What should we do?

**Comment by rasnesakam** [comment]
And also, I am considering to adding [Auth service proto file](https://github.com/rika-hq/rika-auth) as a submodule under the `src/main/proto`. But it looks like that we need add somethings on protofile like:
`option java_multiple_files = true`
and maybe more javaish package name like:
`package org.rika.services.auth` or `package org.rika.auth` or `com.rika.auth` etc.
As  because of our project group name is `com.rika`, I think it's good to name it like `com.rika.auth`or `com.rika.services.auth`

**Comment by mkaramuk** [comment]
> And also, I am considering to adding [Auth service proto file](https://github.com/rika-hq/rika-auth) as a submodule under the `src/main/proto`. But it looks like that we need add somethings on protofile like: `option java_multiple_files = true` and maybe more javaish package name like: `package org.rika.services.auth` or `package org.rika.auth` or `com.rika.auth` etc. As because of our project group name is `com.rika`, I think it's good to name it like `com.rika.auth`or `com.rika.services.auth`

Yes I think we'll need to add Java options to proto files (like `option java_multiple_files`).

We should left `package` as it is (it will be used by another services which written in another languages). Because there is another option to define Java package name. We don't need to change `package`.

Could you examine and determine which options should be added to protofile by using [protobuf doc](https://protobuf.dev/getting-started/javatutorial/) and open issue at [auth service proto](https://github.com/rika-hq/rika-auth-proto) repository?

**Comment by mkaramuk** [comment]
> After done some research and lookup, I found gradle file that we want [here](https://github.com/grpc/grpc-java/blob/master/examples/build.gradle). After applying this configuration, project defines proto file under main module. <img alt="image" width="236" src="https://user-images.githubusercontent.com/45129228/280964981-95939c91-2e30-4356-9d0b-dd0d03830ee8.png"> When I look at to [this video](https://www.youtube.com/watch?v=DU-q5kOf2Rc) Guy created proto folder inside main module, and defined it as a resource folder. Our is defined as source folder. This configuration also implements some tasks for proto generating like below, But I didn't tested it <img alt="image" width="271" src="https://user-images.githubusercontent.com/45129228/280968062-59a902f5-de7e-4eb7-95eb-65b53d75a330.png"> In my opinion, this case covers issue #3 . What should we do?

I didn't watch the complete video. But one of the projects the guy created is gRPC server and other one is client (the one in the resources folder I think). We need to generate files as client.

As well as it could be generated by using gradle which covered by #3, it doesn't matter where the submodule is. It could be under resources or somewhere else.
